api/swagger.yml

```diff
@@ -1527,6 +1530,36 @@ paths:
         default:
           $ref: "#/components/responses/ServerError"
 
+  /read_only_repositories:
+    post:
+      tags:
+        - internal
+      operationId: createReadOnlyRepository
+      summary: create read-only repository
```

Comment:
This is a _very_ specific path.  I would much prefer that we nest it.  For instance we might POST to `/internal/repositories`, and have a `read_only` flag that is supported only there.

Advantages of putting internal endpoints together:
- Some organization of paths
- Can define load-balancer rules to allow or block these
- Having POST /internal/repositories that is like POST /repositories only with more options allowed is easier to understand and handle.

The chief disadvantage that I can see is:
- `read_only` becomes an option on this (internal-only) endpoint.